pkg/controller/projects/accesstokens/controller.go

```diff
@@ -0,0 +1,237 @@
+limitations under the License.
+*/
+
+package accesstokens
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/xanzy/go-gitlab"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane-contrib/provider-gitlab/apis/projects/v1alpha1"
+	"github.com/crossplane-contrib/provider-gitlab/pkg/clients"
+)
+
+const (
+	errNotAccessToken             = "managed resource is not a Gitlab accesstoken custom resource"
+	errGetFailed                  = "cannot get Gitlab accesstoken"
+	errCreateFailed               = "cannot create Gitlab accesstoken"
+	errDeleteFailed               = "cannot delete Gitlab accesstoken"
+	errProjecAccessTokentNotFound = "cannot find Gitlab accesstoken"
+)
+
+// SetupAccessToken adds a controller that reconciles ProjectAccessTokens.
+func SetupAccessToken(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.AccessTokenKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.AccessToken{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.AccessTokenGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newGitlabClientFn: newAccessTokenClient}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube              client.Client
+	newGitlabClientFn func(cfg clients.Config) gitlab.ProjectAccessTokensService
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.AccessToken)
+	if !ok {
+		return nil, errors.New(errNotAccessToken)
+	}
+	cfg, err := clients.GetConfig(ctx, c.kube, cr)
+	if err != nil {
+		return nil, err
+	}
+	return &external{kube: c.kube, client: c.newGitlabClientFn(*cfg)}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client gitlab.ProjectAccessTokensService
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.AccessToken)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotAccessToken)
+	}
+
+	externalName := meta.GetExternalName(cr)
+	if externalName == "" {
+		return managed.ExternalObservation{ResourceExists: false}, nil
```

Comment:
`false` is the default value.

```suggestion
		return managed.ExternalObservation{}, nil
```